bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js

```diff
@@ -222,11 +222,32 @@ pimcore.object.tags.advancedManyToManyObjectRelation = Class.create(pimcore.obje
                     }
                 }
             } else if (this.fieldConfig.columns[i].type == "bool") {
+
+                let defaultVal = this.fieldConfig.columns[i].value;
+                let keyType = this.fieldConfig.columns[i].type;
+                let keyName = this.fieldConfig.columns[i].key;
+                
                 renderer = function (value, metaData, record, rowIndex, colIndex, store) {
                     if (this.fieldConfig.noteditable) {
                         metaData.tdCls += ' grid_cbx_noteditable';
                     }
 
+                    //Check to see if the "value" variable has had a value assigned 
+                    //to it. If not, it means that this is the first time the variable
+                    //is being used, and therefore we will look to see if there was
+                    //a default value assigned, and if so, take the appropriate action.
+                    if(value != false && value != true) {
+                        if(defaultVal == 'true' || defaultVal == 'false') {
+                            if(defaultVal == 'true')
+                                value = true;
+                            else
+                                value = false; 
+
+                            record.set(keyType, value);
+                            record.set(keyName, value);
+                        }
+                    }
```

Comment:
Seems to use the same logic as below, maybe they have a helper file we can move this too so it can be reused.